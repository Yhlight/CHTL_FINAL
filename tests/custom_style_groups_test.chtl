// Test file for [Custom] @Style groups

// 1. Base template
[Template] @Style Base {
    font-size: 16px;
    color: black;
}

// 2. Custom template with placeholders
[Custom] @Style CustomText {
    font-family; // placeholder
    line-height: 1.6;
    color; // placeholder
}

// 3. Custom template inheriting and specializing another
[Custom] @Style WarningText {
    @Style Base { // Inherits from Base
        delete color; // Deletes a property from the inherited template
    }
    font-style: italic; // Adds a new property
    color: orange; // Overrides a property (though it was deleted, this adds it back)
}

// 4. Using the templates

html {
    body {
        // Test placeholder fulfillment
        p {
            style {
                @Style CustomText {
                    font-family: "Arial";
                    color: blue;
                }
            }
            text: "This text uses a custom template with placeholders.";
        }

        // Test inheritance and specialization
        p {
            style {
                @Style WarningText;
            }
            text: "This is a warning text.";
        }

        // Test usage-time delete
        p {
            style {
                @Style WarningText {
                    delete font-style;
                }
            }
            text: "This is a warning text with one style deleted.";
        }
    }
}
