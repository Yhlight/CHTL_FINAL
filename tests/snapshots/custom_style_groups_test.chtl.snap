<html>
  <body>
    <p style="color: blue; font-family: Arial; line-height: 1.6; ">
      This text uses a custom template with placeholders.
    </p>
    <p style="color: orange; font-size: 16px; font-style: italic; ">
      This is a warning text.
    </p>
    <p style="color: orange; font-size: 16px; ">
      This is a warning text with one style deleted.
    </p>
  </body>
</html>
